cs/di-extensions.texy

```diff
@@ -85,61 +127,82 @@ services:
 		factory: MyBlog\Components\ArticlesList(@extension.articles)
 \--
 
-Který načteme a dodatečně služby nastavíme
+Který načteme a dodatečně služby nastavíme.
 
 /--php
-public function loadConfiguration()
+class BlogExtension extends Nette\DI\CompilerExtension
 {
-	$this->config += $this->defaults;
-	$builder = $this->getContainerBuilder();
-
-	// načtení konfiguračního souboru pro rozšíření
-	$this->compiler->loadDefinitions($builder, $this->loadFromFile(__DIR__ . '/blog.neon')['services'], $this->name);
-
-	// počet článků na stránku v komponentě
-	$builder->getDefinition($this->prefix('articlesList'))
-		->addSetup('setPostsPerPage', [$this->config['postsPerPage']]);
-
-	// volitelné vypnutí komentářů
-	if (!$this->config['comments']) {
-		$builder->getDefinition($this->prefix('comments'))
-			->addSetup('disableComments');
+	public function loadConfiguration()
+	{
+		// načtení konfigurační parametrů
+		$config = $this->validateConfig($this->defaults);
+		$builder = $this->getContainerBuilder();
+
+		// načtení konfiguračního souboru pro rozšíření
+		$this->compiler->loadDefinitions($builder, $this->loadFromFile(__DIR__ . '/blog.neon')['services'], $this->name);
+
+		// počet článků na stránku v komponentě
+		$builder->getDefinition($this->prefix('articlesList'))
+			->addSetup('setPostsPerPage', [$config['postsPerPage']]);
+
+		// volitelné vypnutí komentářů
+		if (!$config['comments']) {
+			$builder->getDefinition($this->prefix('comments'))
+				->addSetup('disableComments');
+		}
 	}
 }
 \--
 
 
-CompilerExtension::beforeCompile() .{toc: beforeCompile()}
-==========================================================
+beforeCompile() .[method]
+=========================
 
-V této fázi sestavování už by neměly přibývat další služby. Můžeme ovšem upravovat existující a doplnit některé potřebné vazby mezi službami, například pomocí tagů.
+V této fázi sestavování už by neměly přibývat další služby. Můžeme ovšem upravovat existující a doplnit některé potřebné vazby mezi službami, například pomocí tagů (`findByTag()`) nebo podle typů (`findByType()`).
```

Comment:
Metoda se volá ve chvíli, kdy kontejner obsahuje všechny služby přidané jednotlivými rozšířeními v metodách `loadConfiguration` a taktéž uživatelskými konfiguračními soubory. V této fázi sestavování tedy můžeme definice služeb upravovat nebo doplnit vazby mezi nimi. Pro vyhledávání služeb v kontejneru podle tagů lze využít metodu `findByTag()`, podle třídy či rozhraní zase metodu `findByType()`.